ip/flac.c

```diff
@@ -427,11 +415,14 @@ static int flac_seek(struct input_plugin_data *ip_data, double offset)
 
 	sample = (uint64_t)(offset * (double)sf_get_rate(ip_data->sf) + 0.5);
 	if (!F(seek_absolute)(priv->dec, sample)) {
+		if (F(get_state(priv->dec)) == FLAC__STREAM_DECODER_SEEK_ERROR) {
+			if (!F(flush)(priv->dec))
+				d_print("failed to flush\n");
+			priv->buf_rpos = 0;
+			priv->buf_wpos = 0;
+		}
 		return -IP_ERROR_ERRNO;
 	}
-	priv->ignore_next_write = 1;
-	priv->buf_rpos = 0;
-	priv->buf_wpos = 0;
```

Comment:
No. These should never be reset when things are performing without error. otherwise you might get partial samples if flac_read is not finished through the end, and the stream will be broken.